File: content/problems/3106-lexicographically-smallest-string-after-operations-with-constraint/solutions/left-to-right-cyclic-distance-greedy/solution.cpp
class Solution {
    public:string getSmallestString(string s,int k){
        for(char& letter:s){
            if(k==0)break;
            int direct=letter-'a',distance=min(direct,26-direct);
            if(distance<=k){
                letter='a';
                k-=distance;
            }else{
                letter-=k;
                k=0;
            }
        }
        return s;
    }
};

File: content/problems/3106-lexicographically-smallest-string-after-operations-with-constraint/solutions/left-to-right-cyclic-distance-greedy/Solution.java
class Solution {
    public String getSmallestString(String s,int k){
        char[] letters=s.toCharArray();
        for(int index=0;index<letters.length&&k>0;index++){
            int direct=letters[index]-'a',distance=Math.min(direct,26-direct);
            if(distance<=k){
                letters[index]='a';
                k-=distance;
            }else{
                letters[index]=(char)(letters[index]-k);
                k=0;
            }
        }
        return new String(letters);
    }
}
